### ISSUE TITLE
astutil.py line 68 `name = child.targets[0].value` "'Name' object has no attribute 'value'" (Py 3.4, 3.5)

### ISSUE THREAD

**Issue by Twangist** [alex-sherman/deco#17]
On trying to adapt an existing program to use deco, I'm getting an error on the first (& only) call to the `synchronized` function:

    my_fn( _list_of_dicts, _dict, _int1, _int2, _str )

with args of the indicated types. Traceback starting from that call is:

    File "~/_VENVs/resizer-with-deco/lib/python3.5/site-packages/deco/conc.py", line 47, in __call__
      rewriter.visit(node.body[0])
    File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/ast.py", line 245, in visit
      return visitor(node)
    File "~/_VENVs/resizer-with-deco/lib/python3.5/site-packages/deco/astutil.py", line 85, in visit_FunctionDef
      self.generic_visit(node)
    File "~/_VENVs/resizer-with-deco/lib/python3.5/site-packages/deco/astutil.py", line 55, in generic_visit
      super(NodeTransformer, self).generic_visit(node)
    File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/ast.py", line 253, in generic_visit
      self.visit(item)
    File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/ast.py", line 245, in visit
    return visitor(node)
    File "~/_VENVs/resizer-with-deco/lib/python3.5/site-packages/deco/astutil.py", line 55, in generic_visit
      super(NodeTransformer, self).generic_visit(node)
    File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/ast.py", line 253, in generic_visit
      self.visit(item)
    File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/ast.py", line 245, in visit
      return visitor(node)
    File "~/_VENVs/resizer-with-deco/lib/python3.5/site-packages/deco/astutil.py", line 68, in generic_visit
      name = child.targets[0].value
    AttributeError: 'Name' object has no attribute 'value'

I haven't yet stripped things down to a more basic example.

**Comment by alex-sherman** [comment]
This would be a great case to have a better error message, thanks for pointing it out!

What's happening is you're trying to assign the result of the `@concurrent` function straight to a variable like:
```python
result = concurrent(...)
```
This is not allowed in `@synchronized` functions, I had updated the readme recently to explain this a little. You'll need to assigned to an indexable object like:
```python
result[key] = concurrent(...)
```
I'll make a note to have this use case generate a more useful error.

Out of curiosity, could you post the body of your `@synchronized` function? I would be interested to see if instead this is a case that we should make deco handle. Thanks very much!

**Comment by Twangist** [comment]
You're welcome. Thank you very much for an elegant and deep package. 
I got past the problem -- in fact, I have things working now. The problem was as you suspected, and it's amusing because I was assigning the value of the concurrent function to `_`.  My @synchronized function is just a simple loop through a list `lst`, calling the concurrent function for each index:

        for idx in range(len(data)):
            do_one(lst, idx, ...)    # formerly _ = do_one(<args>)

Originally it used the `tdqm` progress bar, but that turns out to be not multiprocess-friendly. I tried updating the progressbar in `do_one`, with a passed argument `do_one(lst, idx, ..., progressbar)`, but part of the state of progressbar is a closure, which can't be pickled. Updating the progress bar at the end of the loop above has other problems, and anyway I expect 100% would be reached before any actual work got done.
